docs/concepts.md

```diff
@@ -0,0 +1,85 @@
+# ddc concepts
+
+<!--
+Copyright (C) The ddc development team, see COPYRIGHT.md file
+
+SPDX-License-Identifier: MIT
+-->
+
+ddc introduces labels in the form of dimensions, and attributes on top of Kokkos views, which allows for a more intuitive, and less error-prone developer experience.
+
+In fact, in Kokkos, the indices of views are weakly typed, meaning that each index is a simple integer. Let's consider a multidimensional view intended to represent a physical quantity, such as velocity in our example. The first index represents the velocity along the X-axis, and the second index represents the velocity along the Y-axis. In reality, there is nothing to distinguish between these two indices from the computer's perspective; they are both simply integers. This can lead to a situation where one mistakenly swaps the two indices. The code would compile successfully, but the resulting behavior would be incorrect, and the source of the error could be difficult to trace.
+
+The advantage of using ddc is that it provides array-like containers that have labeled dimensions using strongly typed indices. For instance, by labeling dimensions by `X` and `Y`, the indices along those labeled dimensions become strongly typed preventing the user from making such mistakes.
+
+> Note that the use of ddc is not restricted to solving equations. Indeed, one can easily imagine strong typing of variables corresponding to names or ages in a registry. The operations must then be adapted accordingly. Here, we largely base our approach on the uniform and non-uniform resolution of the heat equation in two dimensions, which is why we focus the presentation on solving equations using finite differences on a 2D grid.
+
+## ddc::Chunk and ddc::ChunkSpan
+
+The `ddc::chunk` is a container that holds the data, while `ddc::chunkspan` behaves like `mdspan`, meaning they are pointers to the data contained within the chunk.
+
+As mentioned in the introduction, ddc is a library that offers strongly typed indexing. Chunks contain data that can be located in space by coordinates. Thus, to access the data at a specific point in a 2D space, instead of entering two integers corresponding to the x and y positions, ddc requires entering the  coordinate `x` as a discrete element of the x position: `ddc::DiscreteElement<DDimX>`, and entering the `y` coordinate as a discrete element following the y dimension: `ddc::DiscreteElement<DDimY>`. This is done after predefining a strong typing for the discretized `X` dimension as `DDimX` and a strong typing for the discretized `Y` dimension as `DDimY` (see the heat equation example \subpage uniform_heat_equation "examples/uniform_heat_equation.cpp").
```

Comment:
```suggestion
  As mentioned in the introduction, DDC is a library that offers strongly typed indexing. Chunks contain data that can be accessed by unique identifiers called discrete elements. Thus, to access the data at a specific point in a 2D space, instead of entering two integers corresponding to the 'x' and 'y' axes, DDC requires entering the coordinate `x` as a discrete element of the x position: `ddc::DiscreteElement<DDimX>`, and entering the `y` coordinate as a discrete element following the y dimension: `ddc::DiscreteElement<DDimY>`. This is done after predefining a strong typing for the discretized `X` dimension as `DDimX` and a strong typing for the discretized `Y` dimension as `DDimY` (see the heat equation example \subpage uniform_heat_equation "examples/uniform_heat_equation.cpp").
```

I am still not convinced we should talk about discretization in this page.